### DISCUSSION TITLE
Completeness

### DISCUSSION THREAD

**Discussion by ghost** [terraform-coop/terraform-provider-foreman#139] [General]
Hello everyone,

I'm very interested in using the foreman provider for Terraform.
But my question is how "complete" this provider is?

Can I completly configure foreman. Or are there yet still missing some parts?

Thanks

**Comment by togoschi** [comment]
Hi @Hentheroot,

at first i want to point out that this provider isn't focused on "configure foreman" itself but on "manage foreman resources" with infrastructure-as-code. However your question is possibly the seed for the beginning of a better overview of what is already implemented and what is still missing in the provider yet.

I'm an ex-colleague of @lhw at HanseMerkur and i've been deeply involved in the continous development of the provider since he initially forked the provider from https://github.com/wayfair to extend it for the deployment of bare-metal hosts. We at HanseMerkur still not only using the provider heavily for the deployment of our bare-metal hosts and virtual machines but we also donate some development work on new features and bug fixing as well as trying to promote the usage of the provider in the community.

Here is an (short) overview of feature completeness of the provider including the current status [^1]

| API resource | available | status |
| --- | --- | --- |
| foreman common_parameters | x | production-ready [^2] |
| foreman compute_attributes | x | production-ready |
| foreman compute_profiles | x | new feature (PR #134) | 
| foreman compute resources | x |  | 
| foreman domains | x | production-ready |
| foreman hostgroups | x | production-ready |
| foreman hosts | x | production-ready |
| foreman http_proxies | x |   | 
| foreman images | x | fixed recently (PR #133) |
| foreman interfaces | x | production-ready (please note issue #126)
| foreman job_templates | x | not yet released (PR #138) | 
| foreman locations |  |  |
| foreman media | x | production-ready |
| foreman models | x | production-ready |
| foreman operatingsystems | x | production-ready |
| foreman os_default_templates | x |  |
| foreman paramters | x | production-ready | 
| foreman provision templates | x | production-ready |
| foreman ptables | x | production-ready |
| foreman puppetclasses | x |  |
| foreman realms | | |
| foreman settings | x | new feature (PR #120) | 
| foreman smart_proxies | x |  | 
| foreman subnets | x | production-ready | 
| foreman tasks |  |  |
| foreman template_inputs | x | not yet released (PR #138) |
| foreman template_kinds | x | production-ready |
| foreman usergroups | x |  |
| foreman users | x |  |
| katello content_credentials | x |  | 
| katello content views |   | issue #140  |
| katello errata |  |  | 
| katello host_collections |  |  |
| katello lifecycle environments |  | issue #141  |
| katello packages |  |  | 
| katello products | x | latest fix not yet released (PR #142) |
| katello repositories | x | latest fix not yet released (PR #143) |
| katello subscriptions | | | 
| katello sync_plans | x | latest fix not yet released (PR #143) |

[^1]: from my point of view
[^2]: "production-ready" means that we used this kind of terraform resources very often in our production environment
